pkg/webui/console/views/application-general-settings/index.js

```diff
@@ -254,14 +277,20 @@ export default class ApplicationGeneralSettings extends React.Component {
     // If there is a change in attributes, copy all attributes so they don't get
     // overwritten.
     const update =
-      'attributes' in changed ? { ...changed, attributes: appValues.attributes } : changed
+      'attributes' in changed
+        ? {
+            ...changed,
+            attributes: appValues.attributes,
+          }
+        : changed
 
     const {
       ids: { application_id },
     } = application
 
     try {
       await updateApplication(application_id, update)
+      await updateApplicationLink(application_id, update)
```

Comment:
You need to distinguish `update` so that only the relevant fields are included. If you send the whole patch to the `updateApplicationLink` rpc you will get an error because you use invalid field masks.
```suggestion
      const { skip_payload_crypto, ...applicationUpdate } = update
      const linkUpdate = { skip_payload_crypto }
      await updateApplication(application_id, applicationUpdate)
      await updateApplicationLink(application_id, linkUpdate)
```